docs/vector-library.md

```diff
@@ -0,0 +1,123 @@
+# Vector library
+
+## Summary
+
+Implement a standard library that provides functionality for the vector type.
+
+## Motivation
+
+Currently, vectors are a primitive type implemented internally. All of the heavy work to implement vectors is done, but there is no runtime-agnostic way to use vectors in Luau code. Individual Luau runtimes, such as [Lune](https://github.com/lune-org/lune) or Roblox, must provide their own distinct libraries to work with native vectors which may be inconsistent. In cross-runtime code, this results in performance drawbacks & difficulty utilizing native vectors.
+
+## Design
+
+The default metatable for vectors should now be the `vector` library. While buffers and coroutines do not have this, vectors likely should for ergonomics.
+
+### Library functions & constants
+
+This RFC proposes the following basic functions & constants as a starting point, but as with all builtin libraries, future RFCs can propose additional functions.
+
+---
+
+`vector(x: number, y: number, z: number)`
+
+Creates a vector with 3 components: x, y, z. If the feature flag for wide vectors is enabled, a fourth argument `w: number?` will be introduced.
+
+Due to the common usage of vectors, vector creation should be ergonomic. Therefore, it is probably worth breaking the `create()` naming standard.
+
+`vector.magnitude(vecA: vector): number`
+
+Calculates the magnitude of a given vector.
+
+`vector.normalized(vec: vector): vector`
+
+Returns the normalized version (aka unit vector) of a given vector. If a zero vector is passed, return zero. For further clarification, `vector.normalized(vector(0, 0, 0))` should return zero.
```

Comment:
For compatibility with Roblox runtime and for code size and performance (including branchless native codegen lowering) I suggest removing requirements for zero length vector detection.

Reply:
That makes sense.